### PR TITLE
split out index reference tests into their atomic units

### PR DESCRIPTION
**Pull request by czue**
I can't figure out what's causing the below build fails to happen intermittently, but that test was horrible anyways, so maybe splitting it up will either make it more clear or fix it.

@NoahCarnahan or whoever. tests only

```
======================================================================
FAIL: testIndexReferences (casexml.apps.phone.tests.test_sync_mode.FileRestoreSyncTokenUpdateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 45, in inner
    return helper(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 32, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/tzmigration/test_utils.py", line 28, in call_with_settings
    fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 28, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 298, in testIndexReferences
    child_id: [new_index_ref]})
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 108, in _testUpdate
    self._checkLists(index_ids, sync_log.index_tree.indices[case_id].values())
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 93, in _checkLists
    self.assertEqual(set(l1), set(l2))
AssertionError: Items in the second set but not the first:
u'updated_mommy_id'
----------------------------------------------------------------------
```

### CODE REVIEW

**Comment by czue** [comment]
(wait for build)

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py [hunk 1]
```diff
@@ -219,84 +219,133 @@ def testOwnUpdatesDontSync(self):
...
-        
-        # create the child        
-        child_id = "baby"
-        index_id = 'my_mom_is'
+        child_id, parent_id, index_id, parent_ref = self._initialize_parent_child()
+        # update the child's index (parent type)
+        updated_type = "updated_type"
+        child = CaseBlock(create=False, case_id=child_id, user_id=USER_ID, version=V2,
+                          index={index_id: (updated_type, parent_id)},
+        ).as_xml()
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py]: E124 closing bracket does not match visual indentation

File: corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py [hunk 2]
```diff
@@ -219,84 +219,133 @@ def testOwnUpdatesDontSync(self):
...
+        """
+        Test that adding multiple indices works as expected
+        """
+        child_id, parent_id, index_id, parent_ref = self._initialize_parent_child()
         # add new index
-        new_index_id = "my_daddy"
-        new_index_type = "dad"
+        new_case_id = 'new_case_id'
+        new_index_id = 'new_index_id'
+        new_index_type = 'new_index_type'
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py]: F841 local variable 'new_index_type' is assigned to but never used

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**